src/compute/src/render/reduce.rs

```diff
@@ -508,8 +541,12 @@ where
                         continue;
                     }
                     let message = "Non-positive multiplicity in DistinctBy Retractions";
-                    warn!(?row, ?count, debug_name, "[customer-data] {message}");
```

Comment:
One downside to this approach is that we lose structured field logging.  How ugly would it be to supply some macros that take an `ErrorLogger`?

Reply:
I've not been able to come up with anything ergonomic so far. We could probably introduce something like:

```rust
log_dataflow_error!(
    self.error_logger(),
    "Non-positive multiplicity in DistinctBy Retractions",
    ?row, ?count, debug_name,
    "[customer-data] Non-positive multiplicity in DistinctBy Retractions",
);
```

Doesn't seem too bad, actually. Especially because we could make it possible to reference the message in the details format string, and make the `debug_name` be implicitly added to all logs.

```rust
log_dataflow_error!(
    self.error_logger(),
    "Non-positive multiplicity in DistinctBy Retractions",
    ?row, ?count,
    "[customer-data] {message}",
);
```

I'm just not sure whether my macro foo is up to this task :)

In any case, I'd rather make this the scope of another PR, since I foresee some bikeshedding.

Reply:
Nice!   I'm also happy to do this if you don't want to deal with it.